apps/sens/fsw/unit_test/sens_cmds_test.cpp

```diff
@@ -48,43 +51,239 @@
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "Recvd NOOP. Version %d.%d.%d.%d",
+                            SENS_MAJOR_VERSION, SENS_MINOR_VERSION,
+                            SENS_REVISION, SENS_MISSION_REV);
+
+    UtAssert_EventSent(SENS_CMD_NOOP_EID, CFE_EVS_INFORMATION, expectedEvent,
+                       "ProcessCmdPipe, Noop Event Sent");
+    UtAssert_True(oSENS.HkTlm.usCmdCnt == 1, "ProcessCmdPipe, Noop");
 }
 
 
-int32 SENS_Cmds_Test_UT_CFE_SB_SubscribeHook2(CFE_SB_MsgId_t MsgId, CFE_SB_PipeId_t PipeId,
-                                                CFE_SB_Qos_t Quality, uint16 MsgLim)
+/**
+ * Test SENS ProcessCmdPipe, Reset
+ */
+void Test_SENS_ProcessCmdPipe_Reset(void)
 {
-    return 6;
+    SENS              oSENS;
+
+    int32             CmdPipe;
+    SENS_NoArgCmd_t   InMsg;
+
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    CmdPipe = Ut_CFE_SB_CreatePipe("SENS_CMD_PIPE");
+    CFE_SB_InitMsg ((void*)&InMsg, SENS_CMD_MID, sizeof(InMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&InMsg, (uint16)SENS_RESET_CC);
+    Ut_CFE_SB_AddMsgToPipe((void*)&InMsg, (CFE_SB_PipeId_t)CmdPipe);
+
+    SENS_Test_PrintCmdMsg((void*)&InMsg, sizeof(SENS_NoArgCmd_t));
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SUCCESS, 1);
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_GETMSGID_INDEX, SENS_SEND_HK_MID, 1);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
 }
 
 
-void SENS_Function2_Test_Case1(void)
+/**************************************************************************
+ * Tests for SENS VerifyCmdLength()
+ **************************************************************************/
+/**
+ * Test SENS VerifyCmdLength(), Fail CmdLength
+ */
+void Test_SENS_VerifyCmdLength_Fail_CmdLength(void)
 {
-/*    int32 Result;
+    SENS              oSENS;
+
+    bool              bResult = TRUE;
+    bool              bExpected = FALSE;
+    SENS_NoArgCmd_t   CmdMsg;
+    char              expectedEvent[CFE_EVS_MAX_MESSAGE_LENGTH];
 
-    Variable3 = 3;
+    CFE_SB_InitMsg ((void*)&CmdMsg, SENS_CMD_MID, sizeof(CmdMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&CmdMsg, (uint16)SENS_RESET_CC);
+
+    SENS_Test_PrintCmdMsg((void*)&CmdMsg, sizeof(SENS_NoArgCmd_t));
 
-    Ut_CFE_SB_SetFunctionHook(UT_CFE_SB_SUBSCRIBE_INDEX, &SENS_Cmds_Test_UT_CFE_SB_SubscribeHook1);
-*/
     /* Execute the function being tested */
-/*    Result = SENS_Function2();*/
-    
+    oSENS.InitData();
+    bResult = oSENS.VerifyCmdLength((CFE_SB_MsgPtr_t)&CmdMsg, sizeof(CmdMsg) + 5);
+
     /* Verify results */
-/*    UtAssert_True (Variable4 == 4, "Variable4 == 4");
-    UtAssert_True (Result == 25, "Result == 25");
+    sprintf(expectedEvent, "Rcvd invalid msgLen: msgId=0x%08X, cmdCode=%d, "
```

Comment:
PERFECT!   That is exactly how these unit tests should be written.  I don't think I could have written it any better.